python/sqlflow_submitter/tensorflow/predict.py

```diff
@@ -183,13 +184,26 @@ def add_to_example(example, x, i):
                 example.features.feature[feature_name].int64_list.value.extend(
                     list(values))
         else:
-            idx = feature_column_names.index(feature_name)
             if "feature_columns" in feature_columns:
+                idx = feature_column_names.index(feature_name)
                 fc = feature_columns["feature_columns"][idx]
             else:
-                # FIXME(typhoonzero): when the model is deep and wide model, need to
-                # find the feature column form column name, use type numeric_column for now.
-                fc = tf.feature_column.numeric_column("tmp")
+                # DNNLinearCombinedXXX have dnn_feature_columns and leaner_feature_columns param.
+                idx = -1
+                try:
+                    idx = feature_column_names_map[
+                        "dnn_feature_columns"].index(feature_name)
+                    fc = feature_columns["dnn_feature_columns"][idx]
+                except:
```

Comment:
It seems there should be two try blocks for `dnn_feature_columns` and `linear_feature_columns` respectively here. The current code will check `linear_feature_columns ` only if `dnn_feature_columns` is not specified.

Reply:
Well, that's expected, for `DNNLinearCombinedXXX` you must specify which columns are for `dnn_feature_columns` and `linear_feature_columns`, both should be specified. If the user wants to use a pure linear model, he can use `LinearXXX` estimators.